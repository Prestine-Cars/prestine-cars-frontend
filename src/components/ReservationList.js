// @ts-nocheck
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchReservations } from '../redux/actions/reservation';
import ReservationCardItem from './ReservationCardItem';
import classes from './modules/CityDetail.module.css';

const ReservationList = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchReservations());
  }, []);

  const { reservations, loading } = useSelector((state) => state.reservation);

  return (
    <section className={`${classes.content_container}`}>
      {loading && (
      <div className="d-flex flex-column justify-content-center align-items-center vh-100">
        <span className={`spinner-border ${classes.green}`} />
        <p>Loading your reservations...</p>
      </div>
      )}
      <div className="text-center my-4 mw-70">
        <h2 className={`${classes.hero_text}`}>These are your Reservations</h2>
      </div>
      <div className={`pt-4 ${classes.car_wrapper}`}>
        <ReservationCardItem reservations={reservations} />
      </div>
    </section>
  );
};

export default ReservationList;
