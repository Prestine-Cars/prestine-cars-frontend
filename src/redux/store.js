import { configureStore } from '@reduxjs/toolkit';
import userReducer from './reducers/user';
import cityReducer from './reducers/city';
import citiesReducer from './reducers/cities';
import carReducer from './reducers/car';
import reservationReducer from './reducers/reservation';
import addCityReducer from './reducers/addCity';
import deleteCityReducer from './reducers/deleteCity';

const store = configureStore({
  reducer: {
    user: userReducer,
    cities: citiesReducer,
    city: cityReducer,
    add_city: addCityReducer,
    delete_city: deleteCityReducer,
    car: carReducer,
    reservation: reservationReducer,
  },
});

export default store;
