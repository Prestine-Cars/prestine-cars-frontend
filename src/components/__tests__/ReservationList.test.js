import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import ReservationList from '../ReservationList';

describe('Test presence of Reservations', () => {
  test('should renderReservations component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <ReservationList />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
