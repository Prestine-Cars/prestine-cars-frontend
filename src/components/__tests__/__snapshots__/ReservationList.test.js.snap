// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Reservations should renderReservations component 1`] = `
<section
  className="content_container"
>
  <div
    className="d-flex flex-column justify-content-center align-items-center vh-100"
  >
    <span
      className="spinner-border green"
    />
    <p>
      Loading your reservations...
    </p>
  </div>
  <div
    className="text-center my-4 mw-70"
  >
    <h2
      className="hero_text"
    >
      These are your Reservations
    </h2>
  </div>
  <div
    className="pt-4 car_wrapper"
  />
</section>
`;
